march_fake_sensor_data/src/FakeTemperatureData.cpp

```diff
@@ -120,7 +120,21 @@ int main(int argc, char** argv)
 
   n.getParam(n.getNamespace() + "/min_temperature", min_temperature);
   n.getParam(n.getNamespace() + "/max_temperature", max_temperature);
-  n.getParam("/sensors", sensor_names);
+
+
+  int count = 0;
+  while (!n.hasParam("/march/joint_names"))
+  {
+    ros::Duration(0.5).sleep();
+    count++;
+    if (count > 10)
+    {
+      ROS_ERROR("Failed to read the joint_names from the parameter server.");
+      throw std::runtime_error("Failed to read the joint_names from the parameter server.");
+    }
+  }
+
+  n.getParam("/march/joint_names", sensor_names);
```

Comment:
Just double checking, since I do not know. But in what order is this code running?
First the while loop for a certain amount of time and then the getParam? If this is the case, I foresee a problem. 
Or is the getParam also executed while the while loop is running? If this is the case, everything is fine.

Reply:
What makes you think this code would run non-deterministically?

The code is executed from top to bottom.